Axis inconsistency

ManevilleF opened the pull request:
> Closes #110 

# Context (#110)

Currently the `Hex` coordinate system follows the convention that `y` is *going down*, for flat hexagons mostly as pointy hexagons it would go *down-right*. This is inconsistent:
- with the 2d Y-up convention of game engines like **Bevy**, this is not a big issue, as the hex coords don't relate directly to pixel coordinates
- with the `Direction` enum, `Direction::Top` is equivalent of `hex(0, -1)` and `Direction::Bottom`hex(0, 1)` which is a major inconsistency with the coordinate system.

 ~~This PR provides a fix with 1e1103cb95fc8074f6c44a27b87e4aa737d82d36 which does the following:
Invert the `y` axis in `HexLayout::hex_to_world_pos` and `HexLayout::world_pos_to_hex`. This is a debatable breaking change~~
This Pr provides a fix with 47ce48105722ee582f0d850257401b2c658262e5 which does the following:
Adds the option to invert each axis in `HexLayout`, keeping the current behaviour as default to avoid breaking change

Other solutions:
- [ ] ~~Keep the *y up* as it is but rename `Direction` variants to correctly represent the orientation~~
- [ ] ~~Keep the *y up* but rename `Direction` variants based on axis instead of orientation~~
- [x] Allow customization in `HexLayout` with something like `y_up: bool` which is probably a good idea

ManevilleF commented:
@uvizhe can you test this PR ? does it fix the #110 correctly ?

uvizhe commented:
Tested it, orientation is good!

Side note: **Bevy** is always Y-down since 0.11.

alice-i-cecile reviewed (approved):
Seems like a fine change. I'd definitely call this breaking though :)

ManevilleF commented:
> Side note: **Bevy** is always Y-down since 0.11.

Yeah for screen coordinates but not for the world coords

ManevilleF commented:
@alice-i-cecile @uvizhe I think I prefer the new solution of 47ce48105722ee582f0d850257401b2c658262e5:
Instead of inverting the `y` axis in `HexLayout` I added the option to invert both, this is no longer a breaking change but will still require a minor version.
This means that the default behaviour is the same but I updated the `hex_grid` example to invert the `y_axis`. @uvizhe do you think that change fixes #110 ?
Should I invert the `y` axis by default still ?

uvizhe commented:
I think Top is top and Bottom is bottom regardless of coordinate system. It's about how we see natural world and how describe it. For almost everybody (I guess 🙃), looking at computer screen, top and up are synonymous as well as bottom and down. So the default should be natural and there is no inconsistency and no room for misinterpretation here.

Having an option to invert axes is totally fine by me.

ManevilleF commented:
Since hexx is mostly used by `bevy` it makes sens to use its standard of y coordinates, therefore the `y` axis will be inverted by default
